lib/instrumentation/fastify.js

```diff
@@ -83,9 +81,33 @@ module.exports = function initialize(agent, fastify, moduleName, shim) {
 
       setupRouteHandler(shim, fastifyForWrapping)
 
-      setupMiddlewareHandlers(shim, fastifyForWrapping)
+      // Don't wrap use() in fastify v3+
+      if (!isv3Plus) {
+        setupMiddlewareHandlers(shim, fastifyForWrapping)
+      }
 
       return fastifyForWrapping
     }
   })
+
+  if (isv3Plus) {
+    setupExports(fastify, wrappedExport)
+  }
+}
+
+/**
+ * module.exports = fastify
+ * module.exports.fastify = fastify
+ * module.exports.default = fastify
+ */
+function setupExports(original, wrappedExport) {
+  wrappedExport.fastify = original.fastify
+
+  if (original.fastify) {
+    wrappedExport.fastify = wrappedExport
+  }
+
+  if (original.default) {
+    wrappedExport.default = wrappedExport
+  }
```

Comment:
Decided to instead go for explicitly setting to match the export style. Might as well be more correct in the meantime. Wasn't much more work.